### ISSUE TITLE
Date not shown when form is rendered

### ISSUE THREAD

**Issue by larhou** [mdn/content#27522]
### MDN URL

https://developer.mozilla.org/en-US/docs/Learn/Server-side/Express_Nodejs/forms/Create_BookInstance_form

### What specific section or headline is this issue about?

View

### What information was incorrect, unhelpful, or incomplete?

If the "Create BookInstance" form render again the date is not shown in the field with the label: "Date when book available:"

The reason for the error is that the formatting of the value in the input field with type="date" must be in the format: "yyyy-MM-dd" ("2023-06-29")

This is the source of the HTML code for the "Date when book available" field:

<label for="due_back">Date when book available:</label><input class="form-control" id="due_back" type="date" name="due_back" value="2023-06-29T00:00:00.000Z">

The attached screen shot show that the entered date is not available upon re-render of the form.
![Screenshot 2023-06-24 at 11 27 20](https://github.com/mdn/content/assets/55982196/55c9a445-3f0b-4873-8e7e-bfc016f9513d)

### What did you expect to see?

The HTML for the "Date when book available" input field should be with a value of the date in the format: "yyyy-MM-dd" ("2023-06-24")

<label for="due_back">Date when book available:</label><input class="form-control" id="due_back" type="date" name="due_back" value="2023-06-29">

The attached screen shot show the page where the date entered by the user is now available on re-render og the page after validation and sanitation.

![Screenshot 2023-06-24 at 11 30 13](https://github.com/mdn/content/assets/55982196/733bfc70-81dc-48f0-b53e-e42f8476ef92)

### Do you have anything more you want to share?

A possible solution could be to use `.toLocaleDateString('en-CA')` to format the date into the proper format (yyyy-mm.dd).

It could be done in the bookinstanceController.js file before parsing the variables to the bookinstance_form.pug template :

```
      res.render("bookinstance_form", {
        title: "Create BookInstance",
        book_list: allBooks,
        selected_book: bookInstance.book._id,
        errors: errors.array(),
        bookinstance: bookInstance,
        **due_back_formatted: req.body.due_back.toLocaleDateString('en-CA'),**
      });

```
In the bookinstance_form.pug file the new variable is referenced:

```
    div.form-group
      label(for='due_back') Date when book available:        
      input#due_back.form-control(type='date' name='due_back' value=(undefined===bookinstance ? '' : **due_back_formatted** ))

```

Maybe the reformatting could also be done in the .pug template through a script. 

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/learn/server-side/express_nodejs/forms/create_bookinstance_form`
* MDN URL: https://developer.mozilla.org/en-US/docs/Learn/Server-side/Express_Nodejs/forms/Create_BookInstance_form
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/learn/server-side/express_nodejs/forms/create_bookinstance_form/index.md
* Last commit: https://github.com/mdn/content/commit/9f69ae043943e7b1b074cacebad56877caf82416
* Document last modified: 2023-05-23T10:18:12.000Z

</details>

**Comment by larhou** [comment]
This is another solution ;-)

Just make the `.toLocaleDateString('en-CA')` formatting in the bookinstance_form.pug file:

```
    div.form-group
      label(for='due_back') Date when book available:        
      input#due_back.form-control(type='date' name='due_back' value=(undefined===bookinstance ? '' : **bookinstance.due_back.toLocaleDateString('en-CA')** ))
```

**Comment by hamishwillee** [comment]
Thank you - fixed in #27556

I added a virtual method to do the conversion in the model. The worked version of the project https://github.com/mdn/express-locallibrary-tutorial/ has had this solution in place since the tutorial was created - no idea how it didn't get documented.
